src/coreclr/vm/i386/stublinkerx86.cpp

```diff
@@ -2434,7 +2434,10 @@ namespace
 {
     gc_alloc_context* STDCALL GetAllocContextHelper()
     {
-        return &t_runtime_thread_locals.alloc_context;
+        // TODO: I'm not sure to understand if it should return the gc_alloc_context or the ee_alloc_context
```

Comment:
It needs to return `ee_alloc_context` to be in sync with the other path in X86EmitCurrentThreadAllocContextFetch

Reply:
> It needs to return `ee_alloc_context` to be in sync with the other path in X86EmitCurrentThreadAllocContextFetch

Thanks: I updated the code
I still don't understand the end of the function:
```
    _ASSERTE(Thread::GetOffsetOfThreadStatic(&t_runtime_thread_locals.alloc_context) < INT_MAX);
    X86EmitAddReg(dstreg, (int32_t)Thread::GetOffsetOfThreadStatic(&t_runtime_thread_locals.alloc_context));
```
Why is it needed to call GetAllocContextHelper before since it should return the same thing. 
Am I missing something obvious?  :^)

Reply:
The method has two orthogonal paths:

https://github.com/dotnet/runtime/blob/0746dd3d07cbecea031dab391a0dbbb657ba20e7/src/coreclr/vm/i386/stublinkerx86.cpp#L2456-L2464 is used on Unix. It calls into the helper to fetch the thread static.

https://github.com/dotnet/runtime/blob/0746dd3d07cbecea031dab391a0dbbb657ba20e7/src/coreclr/vm/i386/stublinkerx86.cpp#L2488-L2494 is used on Windows . It has the thread static fetch inline.